File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        bitsBuck = [[] for _ in range(16)]
        arr.sort()
        for a in arr:
            bitsBuck[a.bit_count()].append(a)
        res = sum(bitsBuck, [])
        return res

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        vector<int> res;
        vector<vector<int>> bitsBuck(16);
        sort(arr.begin(), arr.end());
        for(int i = 0; i < arr.size(); i++)
        {
            bitsBuck[__builtin_popcount(arr[i])].push_back(arr[i]);
        }
        for(int i = 0; i < bitsBuck.size(); i++)
        {
            res.insert(res.end(), bitsBuck[i].begin(), bitsBuck[i].end());
        }
        return res;
    }
};